pkg/frontend/mysql_cmd_executor.go

```diff
@@ -191,6 +193,96 @@ func (o *outputQueue) flush() error {
 	return nil
 }
 
+/*
+handle show create table in plan2 and tae
+*/
+func handleShowCreateTable2(obj interface{}, bat *batch.Batch) (*batch.Batch, error) {
+	ses := obj.(*Session)
+	var vecs []*vector.Vector
+	col := bat.Vecs[0].Col.(*types.Bytes)
+	dbName := string(col.Data[col.Offsets[0] : col.Offsets[0]+col.Lengths[0]])
+
+	col = bat.Vecs[1].Col.(*types.Bytes)
+	tableName := col.Data[col.Offsets[0] : col.Offsets[0]+col.Lengths[0]]
+
+	//get database
+	storage := ses.GetStorage()
+	txnHandler := ses.GetTxnHandler()
+	db, err := storage.Database(dbName, txnHandler.GetTxn().GetCtx())
+	if err != nil {
+		return nil, err
+	}
+
+	//get table
+	table, err := db.Relation(string(tableName), txnHandler.GetTxn().GetCtx())
+	if err != nil {
+		return nil, err
+	}
+
+	//get attributes
+	defs := table.TableDefs(txnHandler.GetTxn().GetCtx())
+	var pkDefs []*engine.PrimaryIndexDef
+	createStr := fmt.Sprintf("CREATE TABLE `%s` (", tableName)
+	rowCount := 0
+	for _, def := range defs {
+		if attr, ok := def.(*engine.AttributeDef); ok {
+			nullOrNot := ""
+			if attr.Attr.Primary {
+				nullOrNot = "NOT NULL"
+			} else {
+				nullOrNot = "DEFAULT NULL"
+			}
+			if rowCount == 0 {
+				createStr += "\n"
+			} else {
+				createStr += ",\n"
+			}
+			typeStr := attr.Attr.Type.String()
+			if attr.Attr.Type.Oid == types.T_varchar {
+				typeStr += fmt.Sprintf("(%d)", attr.Attr.Type.Width)
+			}
+			createStr += fmt.Sprintf("`%s` %s %s", attr.Attr.Name, typeStr, nullOrNot)
+			rowCount++
+		} else if attr2, ok2 := def.(*engine.PrimaryIndexDef); ok2 {
+			pkDefs = append(pkDefs, attr2)
+		}
+	}
+
+	if len(pkDefs) != 0 {
+		for _, def := range pkDefs {
+			pkStr := "PRIMARY KEY ("
+			for _, name := range def.Names {
+				pkStr += fmt.Sprintf("`%s`", name)
+			}
+			pkStr += ")"
+			if rowCount != 0 {
+				createStr += ",\n"
+			}
+			createStr += pkStr
+		}
+	}
+
+	if rowCount != 0 {
+		createStr += "\n"
+	}
+	createStr += ")"
+
+	b := &types.Bytes{}
```

Comment:
You donot need to make a new batch. Just extract the needed columns from the batch after the batch been transformed into the row style.